client-src/overlay/state-machine.js

```diff
@@ -0,0 +1,89 @@
+import { createMachine, assign } from "@xstate/fsm";
```

Comment:
What is size `@xstate/fsm`? Because extra size reduce build time

Reply:
https://bundlejs.com/?q=%40xstate%2Ffsm

It's 3.94kb/1.82kb gzip.

Reply:
Looks like big... Maybe we can reimplement only necessary things or use more small alternative?

Reply:
What's the acceptable size? I can shop around.

Reply:
I just look at `@xstate/fsm`'s [implementation](https://github.com/statelyai/xstate/blob/main/packages/xstate-fsm/src/index.ts), which seems to be quite minimal.

I can think of two options:
1. use `@xstate/fsm` as baseline and continue code golf it to make it as small as possible (probably by removing some error check and some convenient API).
1. pre-minify it before build, but I not sure if this would help the compilation efficiency.

Reply:
I think the `1.` will be better, we can add a new API when we will need it

Reply:
Sorry for no update for more than a month! 😅 

I've just created a custom `createMachine` while maintain enough compatibility so we can use their visualizer.